src/alpakatest/bin/main.cc

```diff
@@ -6,11 +6,12 @@
 #include <filesystem>
 #include <string>
 #include <vector>
+#include "AlpakaCore/alpakaConfig.h"
```

Comment:
@felicepantaleo Ha just one thing: this `alpakatest/bin/main.cc` is actually an old version of `alpaka/bin/main.cc` (the one used for the pixeltrack reconstruction toolchain), where among other aspects I had already addressed this issue of the inclusion of `cuda_runtime.h`.
Maybe that latest version of `main.cc` (the one from `src/alpaka`, https://github.com/cms-patatrack/pixeltrack-standalone/blob/master/src/alpaka/bin/main.cc) could be propagated here, to uniform things, and updated when needed?
The only diff between these 2 files should actually be the `edmodules` / `esmodules`.

If I remember properly, the inclusion of `cuda_runtime.h` is already handled inside `alpaka` anyway, so in a sense, we do not need to re-expose this treatment here? Basically, the choice of the CUDA backend will already trigger the `cuda_runtime.h` inclusion anyway (handled inside `alpaka` library).

Reply:
Ha found it back:
- `DALPAKA_ACC_GPU_CUDA_ENABLED` is set here:
https://github.com/cms-patatrack/pixeltrack-standalone/blob/master/src/alpaka/Makefile#L169 
- `alpaka.hpp` (included at https://github.com/cms-patatrack/pixeltrack-standalone/blob/master/src/alpaka/AlpakaCore/alpakaConfigCommon.h#L4 or similar for the test side) includes `Cuda.hpp`:
https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/alpaka.hpp#L65
- `Cuda.hpp` only includes `cuda_runtime.h` (among others) inside `#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED`: https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/core/Cuda.hpp#L12

Reply:
This also means that for HIP, one might only need to add an extra compilation treatement with a -D ALPAKA_ACC_GPU_HIP_ENABLED, and then the hip_runtime.h will be automatically included as desired (Hip.hpp in alpaka.